### PR TITLE
Infrautils

### PR DESCRIPTION
**Pull request by anneteka**
Added infrautils services to the lighty controller (GuavaCacheProvider, MetricProvide, JobCoordinator)

### CODE REVIEW

File: lighty-core/lighty-controller/pom.xml [hunk 1]
```diff
@@ -47,6 +47,21 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.opendaylight.infrautils</groupId>
+            <artifactId>caches-guava-impl</artifactId>
+            <version>1.6.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.opendaylight.infrautils</groupId>
+            <artifactId>metrics-api</artifactId>
+            <version>1.6.1</version>
```
Comment (vainyksi) [inline lighty-core/lighty-controller/pom.xml]: extract all infrautils versions at least to property...

File: lighty-core/lighty-controller/pom.xml [hunk 2]
```diff
@@ -493,6 +508,30 @@
...
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>findbugs-annotations</artifactId>
+            <version>3.0.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.opendaylight.infrautils</groupId>
+            <artifactId>metrics-impl</artifactId>
+            <version>1.6.1</version>
+        </dependency>
```
Comment (vainyksi) [inline lighty-core/lighty-controller/pom.xml]: move those dependencies above the TEST dependencies

File: lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java [hunk 3]
```diff
@@ -131,6 +140,8 @@
 
     DataBroker getBindingDataBroker();
 
+
+
```
Comment (vainyksi) [inline lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java]: do not introduce unnecessary empty lines

File: lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/impl/LightyControllerImpl.java [hunk 4]
```diff
@@ -184,6 +191,11 @@
     private org.opendaylight.controller.md.sal.binding.impl.BindingDOMNotificationServiceAdapter notificatoinServiceOld;
     private final LightySystemReadyMonitorImpl systemReadyMonitor;
 
+    private CacheProvider cacheProvider;
+    private JobCoordinator jobCoordinator;
+    private MetricProvider metricProvider;
+
+
```
Comment (vainyksi) [inline lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/impl/LightyControllerImpl.java]: do not introduce unnecessary empty lines

File: lighty-core/lighty-controller/pom.xml [hunk 5]
```diff
@@ -493,6 +522,10 @@
             <artifactId>singlenode-configuration</artifactId>
             <scope>test</scope>
         </dependency>
+
+
+
+
```
Comment (vainyksi) [inline lighty-core/lighty-controller/pom.xml]: do not introduce unnecessary empty lines

Comment (anneteka) [reply lighty-core/lighty-controller/pom.xml]: oh no

File: lighty-modules/lighty-infrautils/pom.xml [hunk 6]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>lighty-modules-aggregator</artifactId>
```
Comment (rovarga) [inline lighty-modules/lighty-infrautils/pom.xml]: Never derive from aggregator -- this should be pointing to lighty-parent, in which case ...

File: lighty-modules/lighty-infrautils/pom.xml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                    <target>8</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+    <dependencies>
+        <dependency>
+            <groupId>org.opendaylight.infrautils</groupId>
+            <artifactId>metrics-api</artifactId>
+            <version>1.7.0-SNAPSHOT</version>
```
Comment (rovarga) [inline lighty-modules/lighty-infrautils/pom.xml]: ... you'll get all these imports from there.

File: lighty-modules/lighty-infrautils/pom.xml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>lighty-infrautils</artifactId>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
```
Comment (rovarga) [inline lighty-modules/lighty-infrautils/pom.xml]: this is not appropriate for master -- we require Java 11 tghere

File: lighty-modules/lighty-infrautils/src/main/java/io/lighty/infrautils/LightyCacheProvider.java [hunk 9]
```diff
@@ -0,0 +1,4 @@
+package io.lighty.infrautils;
```
Comment (rovarga) [inline lighty-modules/lighty-infrautils/src/main/java/io/lighty/infrautils/LightyCacheProvider.java]: missing copyright headers

File: lighty-modules/lighty-infrautils/pom.xml (hunk 6 above)
Comment (anneteka) [reply lighty-modules/lighty-infrautils/pom.xml]: this module is deleted in later commits

File: lighty-modules/lighty-infrautils/pom.xml (hunk 7 above)
Comment (anneteka) [reply lighty-modules/lighty-infrautils/pom.xml]: this module is deleted in later commits

File: lighty-modules/lighty-infrautils/pom.xml (hunk 8 above)
Comment (anneteka) [reply lighty-modules/lighty-infrautils/pom.xml]: this module is deleted in later commits

File: lighty-modules/lighty-infrautils/src/main/java/io/lighty/infrautils/LightyCacheProvider.java (hunk 9 above)
Comment (anneteka) [reply lighty-modules/lighty-infrautils/src/main/java/io/lighty/infrautils/LightyCacheProvider.java]: this module is deleted in later commits

**Comment by anneteka** [comment]
@rovarga all changes that were requested reffered to the module that was created and deleted in later commits, so no changes are needed to be done

**Comment by rovarga** [comment]
Okay, can you clean up the series so that each patch is in its final form? Otherwise bisecting on the repo would be ... interesting.

**Comment by anneteka** [comment]
hey, @rovarga I've just noticed that the review is still pending
the changes that were requested (to remove unnecessary commits)were made long ago

File: lighty-core/lighty-controller/pom.xml [hunk 10]
```diff
@@ -15,6 +15,9 @@
         <version>12.0.0-SNAPSHOT</version>
         <relativePath>../lighty-parent</relativePath>
     </parent>
+    <properties>
+        <infrautils-version>1.6.0</infrautils-version>
```
Comment (rovarga) [inline lighty-core/lighty-controller/pom.xml]: This declaration is not needed (and should've been 1.7.0-SNAPSHOT anyway)

File: lighty-core/lighty-controller/pom.xml [hunk 11]
```diff
@@ -47,6 +56,26 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.opendaylight.infrautils</groupId>
+            <artifactId>jobcoordinator-impl</artifactId>
+            <version>${infrautils-version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.opendaylight.infrautils</groupId>
+            <artifactId>caches-guava-impl</artifactId>
+            <version>${infrautils-version}</version>
```
Comment (rovarga) [inline lighty-core/lighty-controller/pom.xml]: remove all <version> declarations here -- version declarations are already correctly set (builds for me)

File: lighty-core/lighty-controller/pom.xml [hunk 12]
```diff
@@ -35,6 +38,12 @@
             <groupId>io.lighty.core</groupId>
             <artifactId>lighty-common</artifactId>
         </dependency>
+        <dependency>
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
+            <version>1.3.2</version>
```
Comment (rovarga) [inline lighty-core/lighty-controller/pom.xml]: why would we need this dependency?

**Pull request closed without merging**